Add null_resource for testing

Msouza91 opened the pull request.

Msouza91 committed:
Add null_resource for testing

Msouza91 committed:
Update main.tf

Msouza91 commented:
atlantis plan

Msouza91 committed:
Update main.tf

Msouza91 commented:
atlantis plan

iac-test-atl commented:
```
Error: This repo is not allowlisted for Atlantis.
```

Msouza91 commented:
atlantis plan

iac-test-atl commented:
Ran Plan for dir: `.` workspace: `default`

```diff
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
+ create

Terraform will perform the following actions:

  # null_resource.atlantis_hw1 will be created
+ resource "null_resource" "atlantis_hw1" {
      + id = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -d .`
* :put_litter_in_its_place: To **delete** this plan click [here](http://20.106.186.209/lock?id=iac-tests%252Fatlantis-example%252F.%252Fdefault)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -d .`

---
* :fast_forward: To **apply** all unapplied plans from this pull request, comment:
    * `atlantis apply`
* :put_litter_in_its_place: To delete all plans and locks for the PR, comment:
    * `atlantis unlock`

Msouza91 committed:
Add template rendering and output

This commit adds a template rendering feature to the main.tf file. It uses the `template_file` data source to render a simple template that prints the content of a variable. The rendered template is then outputted as `rendered_template`.

iac-test-atl commented:
Ran Plan for 2 projects:

1. dir: `.` workspace: `default`
1. dir: `servico` workspace: `default`

### 1. dir: `.` workspace: `default`
<details><summary>Show Output</summary>

```diff
data.template_file.workspace: Reading...
data.template_file.workspace: Read complete after 0s [id=21a3230e03772a58aff1b3709a9e232850916337e1fba95c434076b6668c6e08]

Changes to Outputs:
+ rendered_template = "workspace"

You can apply this plan to save these new output values to the Terraform
state, without changing any real infrastructure.
╷
│ Warning: Version constraints inside provider configuration blocks are deprecated
│ 
│   on providers.tf line 2, in provider "azurerm":
│    2:   version = "=2.0.0"
│ 
│ Terraform 0.13 and earlier allowed provider version constraints inside the
│ provider configuration block, but that is now deprecated and will be
│ removed in a future version of Terraform. To silence this warning, move the
│ provider version constraint into the required_providers block.
╵
```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -d .`
* :put_litter_in_its_place: To **delete** this plan click [here](http://20.106.186.209/lock?id=iac-tests%252Fatlantis-example%252F.%252Fdefault)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -d .`
</details>

---
### 2. dir: `servico` workspace: `default`
**Plan Error**
```
running "/usr/bin/terraform plan -input=false -refresh -out \"/home/atlantis/.atlantis/repos/iac-tests/atlantis-example/4/default/servico/default.tfplan\"" in "/home/atlantis/.atlantis/repos/iac-tests/atlantis-example/4/default/servico": exit status 1
╷
│ Error: No configuration files
│ 
│ Plan requires configuration to be present. Planning without a configuration
│ would mark everything for destruction, which is normally not what is
│ desired. If you would like to destroy everything, run plan with the
│ -destroy option. Otherwise, create a Terraform configuration file (.tf
│ file) and try again.
╵

```

---
* :fast_forward: To **apply** all unapplied plans from this pull request, comment:
    * `atlantis apply`
* :put_litter_in_its_place: To delete all plans and locks for the PR, comment:
    * `atlantis unlock`

Msouza91 committed:
Test atlantis config for workspaces

iac-test-atl commented:
**Plan Error**
```
parsing atlantis.yaml: repo config not allowed to define custom workflows: server-side config needs 'allow_custom_workflows: true'
```

Msouza91 committed:
Refactor atlantis.yaml file to include project names and workflows

iac-test-atl commented:
**Plan Error**
```
parsing atlantis.yaml: repo config not allowed to set 'workflow' key: server-side config needs 'allowed_overrides: [workflow]'
```

Msouza91 commented:
atlantis plan

iac-test-atl commented:
Ran Plan for 0 projects:

Msouza91 committed:
Refactor atlantis.yaml file structure

iac-test-atl commented:
**Plan Error**
```
parsing atlantis.yaml: there are two or more projects with dir: "atlantis-example" workspace: "default" that are not all named; they must have a 'name' key so they can be targeted for apply's separately
```

Msouza91 committed:
Update atlantis.yaml with project names and workspaces

Msouza91 commented:
atlantis plan

iac-test-atl commented:
Ran Plan for 0 projects:

Msouza91 commented:
atlantis plan staging

iac-test-atl commented:
```
Error: unknown argument(s) – staging.
Usage of plan:
  -d, --dir string         Which directory to run plan in relative to root of repo,
                           ex. 'child/dir'.
  -p, --project string     Which project to run plan for. Refers to the name of the
                           project configured in a repo config file. Cannot be used
                           at same time as workspace or dir flags.
      --verbose            Append Atlantis log to comment.
  -w, --workspace string   Switch to this Terraform workspace before planning.
```

Msouza91 commented:
atlantis plan -p servico-staging

iac-test-atl commented:
Ran Plan for dir: `atlantis-example` workspace: `ws_staging`

**Plan Error**
```
dir "atlantis-example" does not exist
```

Msouza91 committed:
Update atlantis.yaml configuration

iac-test-atl commented:
Ran Plan for 2 projects:

1. project: `servico-staging` dir: `.` workspace: `ws_staging`
1. project: `servico-production` dir: `.` workspace: `ws_production`

### 1. project: `servico-staging` dir: `.` workspace: `ws_staging`
<details><summary>Show Output</summary>

```diff
data.template_file.workspace: Reading...
data.template_file.workspace: Read complete after 0s [id=e919a75364398a449f860aeadddc57fa0502145a4e63959ddb33c417a48dc0da]

Changes to Outputs:
+ rendered_template = "staging"

You can apply this plan to save these new output values to the Terraform
state, without changing any real infrastructure.
╷
│ Warning: Version constraints inside provider configuration blocks are deprecated
│ 
│   on providers.tf line 2, in provider "azurerm":
│    2:   version = "=2.0.0"
│ 
│ Terraform 0.13 and earlier allowed provider version constraints inside the
│ provider configuration block, but that is now deprecated and will be
│ removed in a future version of Terraform. To silence this warning, move the
│ provider version constraint into the required_providers block.
╵
```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -p servico-staging`
* :put_litter_in_its_place: To **delete** this plan click [here](http://20.106.186.209/lock?id=iac-tests%252Fatlantis-example%252F.%252Fws_staging)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -p servico-staging`
</details>

---
### 2. project: `servico-production` dir: `.` workspace: `ws_production`
<details><summary>Show Output</summary>

```diff
data.template_file.workspace: Reading...
data.template_file.workspace: Read complete after 0s [id=6754af9632a2745e85c293e5aac0863370d9bd3330b9938c00cadfd215227d77]

Changes to Outputs:
+ rendered_template = "prod"

You can apply this plan to save these new output values to the Terraform
state, without changing any real infrastructure.
╷
│ Warning: Version constraints inside provider configuration blocks are deprecated
│ 
│   on providers.tf line 2, in provider "azurerm":
│    2:   version = "=2.0.0"
│ 
│ Terraform 0.13 and earlier allowed provider version constraints inside the
│ provider configuration block, but that is now deprecated and will be
│ removed in a future version of Terraform. To silence this warning, move the
│ provider version constraint into the required_providers block.
╵
```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -p servico-production`
* :put_litter_in_its_place: To **delete** this plan click [here](http://20.106.186.209/lock?id=iac-tests%252Fatlantis-example%252F.%252Fws_production)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -p servico-production`
</details>

---
* :fast_forward: To **apply** all unapplied plans from this pull request, comment:
    * `atlantis apply`
* :put_litter_in_its_place: To delete all plans and locks for the PR, comment:
    * `atlantis unlock`

Msouza91 committed:
Add repos.yaml file with workflows for staging and production
